Tools/TestWebKitAPI/Tests/WebKitCocoa/Navigation.mm

```diff
@@ -1098,3 +1099,111 @@ HTTPServer server({
     Vector<bool> expectedCallbacks { true, false, false };
     EXPECT_EQ(finishedSuccessfullyCallbacks, expectedCallbacks);
 }
+
+TEST(WKNavigation, HTTPSFirstHTTPDowngradeRedirect)
+{
+    using namespace TestWebKitAPI;
+    HTTPServer httpsServer({
+        { "/"_s, { { }, "hi"_s } }
```

Comment:
I like to add something to the path to avoid confusion and unexpected surprises, like "/secure"